piker/data/elasticsearch.py

```diff
@@ -0,0 +1,108 @@
+# piker: trading gear for hackers
+# Copyright (C) Tyler Goodlet (in stewardship for piker0)
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+'''
+``elasticsearch`` integration.
+
+- client management routines
+- ticK data ingest routines
+- websocket client for subscribing to write triggers
+- todo: tick sequence stream-cloning for testing
+
+'''
+from __future__ import annotations
+from contextlib import asynccontextmanager as acm
+from datetime import datetime
+from pprint import pformat
+from typing import (
+    Any,
+    Optional,
+    Union,
+    TYPE_CHECKING,
+)
+import time
+from math import isnan
+
+from bidict import bidict
+from msgspec.msgpack import encode, decode
+import pyqtgraph as pg
+import numpy as np
+import tractor
+from trio_websocket import open_websocket_url
+import pendulum
+import purerpc
+
+if TYPE_CHECKING:
+    import docker
+    from ._ahab import DockerContainer
+
+from .feed import maybe_open_feed
+from ..log import get_logger, get_console_log
```

Comment:
trying to stick with a multi-line tuple style for any more then one import for a module if that's cool, like above for the `typing` imports.